extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -194,9 +195,22 @@ export class AmpSidebar extends AMP.BaseElement {
     // always create a close button at the end of the sidebar for screen
     // readers.
     element.appendChild(this.createScreenReaderCloseButton());
-    this.registerDefaultAction(invocation => this.open_(invocation), 'open');
-    this.registerAction('toggle', this.toggle_.bind(this));
-    this.registerAction('close', this.close_.bind(this));
+
+    this.registerDefaultAction(invocation => {
+      const {trust, caller} = invocation;
+      this.open_(trust, caller);
```

Comment:
Could these methods (`open`, `close`, and `toggle`) accept an invocation instead? 

This would reduce the script needed here a bit.

Reply:
Hm, most of their callers don't take `ActionInvocation` so we'd need to instantiate them which seems like a bad fit.